docs/2025/puzzles/day09.md

```diff
@@ -2,11 +2,189 @@ import Solver from "../../../../../website/src/components/Solver.js"
 
 # Day 9: Movie Theater
 
+by [@stewSquared](https://github.com/stewSquared)
+
 ## Puzzle description
 
 https://adventofcode.com/2025/day/9
 
+## Solution Summary
+
+We use rectangle representations and search over all possible rectangles for the maximum, filtering for part 2 by checking interesections with boundary lines.
+
+### Part 1
+
+For Part 1, it suffices to calculate the area for all possible rectangles, but modelling this nicely will help with Part 2, so we'll take advantage of some of the tools Scala gives us. We'll use case classes for `Point` and `Area`. An Area is a set representing a rectangular grid of points -- the 3D analog of a [`Range`](https://www.scala-lang.org/api/current/scala/collection/immutable/Range.html). Like a Range, it can function as a virtual collection.
```

Comment:
```suggestion
For Part 1, it suffices to calculate the area for all possible rectangles, but modelling this nicely will help with Part 2, so we'll take advantage of some of the tools Scala gives us. We'll use case classes for `Point` and `Area`. An Area is a set representing a rectangular grid of points -- the 2D analog of a [`Range`](https://www.scala-lang.org/api/current/scala/collection/immutable/Range.html). Like a Range, it can function as a virtual collection.
```